File: packages/audio/android/src/main/java/dev/tcode/thinmpr/audio/model/contract/AlbumModelContract.kt
package dev.tcode.thinmpr.audio.model.contract

import android.net.Uri
import dev.tcode.thinmpr.audio.model.valueObject.AlbumId
import dev.tcode.thinmpr.audio.model.valueObject.ArtistId

interface AlbumModelContract {
    val id: AlbumId
    val name: String
    val artistId: ArtistId
    val artistName: String
    val imageId: String
    val imageUri: Uri
}

File: packages/audio/ios/Model/Contract/AlbumModelContract.swift
import MediaPlayer

protocol AlbumModelContract {
    var media: MPMediaItemCollection { get }

    var id: AlbumId { get }

    var name: String { get }

    var artistId: ArtistId { get }

    var artistName: String { get }

    var imageId: String { get }

    var artwork: MPMediaItemArtwork? { get }
}
